src/Cnab.Bundle/Parameter.cs

```diff
@@ -27,42 +45,65 @@ public class ParameterMetadata
         public string Description { get; set; }
     }
 
-    public class StringParameter : Parameter
+    public class StringParameter : Parameter<string>
     {
-        [JsonProperty("defaultValue")]
-        public string DefaultValue { get; set; }
-
-        [JsonProperty("allowedValues")]
-        public List<string> AllowedValues { get; set; }
-
         [JsonProperty("minLength")]
         public int MinimumLength { get; set; }
 
         [JsonProperty("maxLength")]
         public int MaximumLength { get; set; }
-    }
 
-    public class BoolParameter : Parameter
-    {
-        [JsonProperty("defaultValue")]
-        public bool DefaultValue { get; set; }
+        public override bool IsValid(object value)
+        {
+            if (!(value is string))
+                return false;
 
-        [JsonProperty("allowedValues")]
-        public List<bool> AllowedValues { get; set; }
+            var stringVal = value as string;
+            if (!this.AllowedValues.Contains(stringVal))
+                return false;
+
+            if ((stringVal.Length < this.MinimumLength) || (stringVal.Length > this.MaximumLength))
+                return false;
+
+            return true;
+        }
     }
 
-    public class IntParameter : Parameter
+    public class BoolParameter : Parameter<bool>
     {
-        [JsonProperty("defaultValue")]
-        public int DefaultValue { get; set; }
-
-        [JsonProperty("allowedValues")]
-        public List<int> AllowedValues { get; set; }
+        public override bool IsValid(object value)
+        {
+            if (!(value is bool))
+                return false;
+            
+            return true;
+        }
+    }
 
+    public class IntParameter : Parameter<int>
+    {
         [JsonProperty("minValue")]
         public int MinimumValue { get; set; }
 
         [JsonProperty("maxValue")]
         public int MaximumValue { get; set; }
+
+        public override bool IsValid(object value)
+        {   
+            // TODO - this is rather naive
+            // it should probably handle values such as "80" or 80.0
```

Comment:
Should it?

Reply:
I would suggest throwing some overloads at the problem -
```
public override bool IsValid(int x)
public override bool IsValid(string x)
```

I believe you don't have this in Go, but that doesn't mean this shouldn't be C# idiomatic.

Reply:
I think it _should_ be C# idiomatic, and overloading seems like the best solution.

My main question is the following - should all overloads be available for all parameter types?
Currently, `IsValid`(object value)` is  an abstract method in `IParameter` - should this contain all overloads, forcing all parameter types to accept all overloads?

Reply:
Hm, this might actually make sense, the more I think about it.

Reply:
Updated with overloaded versions for `IsValid` for integer, string and boolean.